testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OIDCProtocolMappersTest.java

```diff
@@ -784,6 +787,46 @@ public void testUserGroupRoleToAttributeMappersScopedClientNotSet() throws Excep
         deleteMappers(protocolMappers);
     }
 
+    @Test
+    public void testUserGroupRoleToAttributeMappersScopedWithDifferentClient() throws Exception {
+        final String clientId = "test-app-scope";
+        final String diffClient = "test-app";
+        final String realmName = "test";
+
+        final ProtocolMapperRepresentation realmMapper = ProtocolMapperUtil.createUserRealmRoleMappingMapper("pref.", "Realm roles mapper", "roles-custom.realm", true, true);
+        final ProtocolMapperRepresentation clientMapper = ProtocolMapperUtil.createUserClientRoleMappingMapper(diffClient, null, "Client roles mapper", "roles-custom.test-app", true, true);
+
+        ClientResource clientResource = ApiUtil.findClientResourceByClientId(adminClient.realm(realmName), clientId);
+        assertThat(clientResource, notNullValue());
+
+        try (ProtocolMappersUpdater protocolMappers = new ProtocolMappersUpdater(clientResource.getProtocolMappers());
+             ClientAttributeUpdater cau = ClientAttributeUpdater.forClient(adminClient, realmName, clientId).setDirectAccessGrantsEnabled(true)) {
```

Comment:
Thank you for the change. You can completely get rid of the `clientResource` this way:
```suggestion
        try (ClientAttributeUpdater cau = ClientAttributeUpdater.forClient(adminClient, realmName, clientId).setDirectAccessGrantsEnabled(true);
          ProtocolMappersUpdater protocolMappers = new ProtocolMappersUpdater(cau.getResource().getProtocolMappers())) {
```